caliban_toolbox/reshape_data.py

```diff
@@ -0,0 +1,703 @@
+
+                # determine if labels are blank, and if so what to do with npz
+                if np.sum(labels) == 0:
+
+                    # blank labels get saved to separate folder
+                    if blank_labels == 'separate':
+                        if verbose:
+                            print('{} is blank, saving to separate folder'.format(npz_id))
+                        save_path = os.path.join(save_dir, blank_labels, npz_id)
+
+                        # save images as either npz or xarray
+                        if save_format == 'npz':
+                            np.savez(save_path + '.npz', X=channels, y=labels)
+
+                        elif save_format == 'xr':
+                            current_xr.to_netcdf(save_path + '.xr')
+
+                    # blank labels don't get saved, empty area of tissue
+                    elif blank_labels == 'skip':
+                        if verbose:
+                            print('{} is blank, skipping saving'.format(npz_id))
+
+                    # blank labels get saved along with other crops
+                    elif blank_labels == 'include':
+                        if verbose:
+                            print('{} is blank, saving to folder'.format(npz_id))
+                        save_path = os.path.join(save_dir, npz_id)
+
+                        # save images as either npz or xarray
+                        if save_format == 'npz':
+                            np.savez(save_path + '.npz', X=channels, y=labels)
+
+                        elif save_format == 'xr':
+                            current_xr.to_netcdf(save_path + '.xr')
+
+                else:
+                    # crop is not blank, save based on file_format
+                    save_path = os.path.join(save_dir, npz_id)
+
+                    # save images as either npz or xarray
+                    if save_format == 'npz':
+                        np.savez(save_path + '.npz', X=channels, y=labels)
+
+                    elif save_format == 'xr':
+                        current_xr.to_netcdf(save_path + '.xr')
+
+    log_data['fov_names'] = fov_names.tolist()
+    log_data['channel_names'] = original_xr.channels.values.tolist()
+    log_data['original_shape'] = original_xr.shape
+    log_data['slice_stack_len'] = resized_xr.shape[1]
+    log_data['save_format'] = save_format
+
+    log_path = os.path.join(save_dir, 'log_data.json')
+    with open(log_path, 'w') as write_file:
+        json.dump(log_data, write_file)
+
+
+def get_saved_file_path(dir_list, fov_name, crop, slice, file_ext='.npz'):
+    """Helper function to identify correct file path for an npz file
+
+    Args:
+        dir_list: list of files in directory
+        fov_name: string of the current fov_name
+        crop: int of current crop
+        slice: int of current slice
+        file_ext: extension file was saved with
+
+    Returns:
+        string: formatted file name
+
+    Raises:
```

Comment:
This doesn't appear to raise an error if a match isn't found, only if too many matches are found